x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -972,6 +978,8 @@ private static Expression processScalarFunction(EsqlScalarFunction f, EsqlFuncti
             boolean childrenChanged = false;
             DataType targetDataType = DataType.NULL;
             Expression arg;
+            DataType targetNumericType = DataType.NULL;
```

Comment:
I think `ImplicitCasting` needs a javadoc to describe in few words its purpose. It initially started with String implicit casting and now it's also doing numeric implicit casting. I believe it will help others looking at the code to pick up faster this class' intention.

Reply:
Will add javadoc.